GitForWindowsHelper/cascading-runs.js

```diff
@@ -133,12 +133,314 @@ const cascadingRuns = async (context, req) => {
 
             return comment
         }
+        if (checkRunOwner === 'git-for-windows'
+            && checkRunRepo === 'git'
+            && name.startsWith('git-artifacts-')) {
+            const output = req.body.check_run.output
+            const match = output.summary.match(
+                /Build Git (\S+) artifacts from commit (\S+) \(tag-git run #(\d+)\)/
+            )
+            if (!match) throw new Error(
+                `Could not parse 'summary' attribute of check-run ${req.body.check_run.id}: ${output.summary}`
+            )
+            const [, ver, commit, tagGitWorkflowRunID] = match
+            const snapshotTag = `prerelease-${ver.replace(/^v/, '')}`
+
+            // First, verify that the snapshot has not been uploaded yet
+            const gitSnapshotsToken = await getToken(context, checkRunOwner, 'git-snapshots')
+            const githubApiRequest = require('./github-api-request')
+            try {
+                const releasePath = `${checkRunOwner}/git-snapshots/releases/tags/${snapshotTag}`
+                await githubApiRequest(
+                    context,
+                    gitSnapshotsToken,
+                    'GET',
+                    `/repos/${releasePath}`,
+                )
+                return `Ignoring ${name} check-run because the snapshot for ${commit} was already uploaded`
+                    + ` to https://github.com/${releasePath}`
+            } catch(e) {
+                if (e?.statusCode !== 404) throw e
+                // The snapshot does not exist yet
+            }
+
+            // Next, check that the commit is on the `main` branch
+            const gitToken = await getToken(context, checkRunOwner, checkRunRepo)
+            const { behind_by } = await githubApiRequest(
+                context,
+                gitToken,
+                'GET',
+                `/repos/${checkRunOwner}/${checkRunRepo}/compare/HEAD...${commit}`,
+            )
+            if (behind_by > 0) {
+                return `Ignoring ${name} check-run because its corresponding commit ${commit} is not on the main branch`
+            }
+
+            const workFlowRunIDs = {}
+            const { listCheckRunsForCommit, queueCheckRun } = require('./check-runs')
+            for (const architecture of ['x86_64', 'i686', 'aarch64']) {
+                const workflowName = `git-artifacts-${architecture}`
+                const runs = name === workflowName ? [req.body.check_run] : await listCheckRunsForCommit(
+                    context,
+                    gitToken,
+                    checkRunOwner,
+                    checkRunRepo,
+                    commit,
+                    workflowName
+                )
+                const needle =
+                    `Build Git ${ver} artifacts from commit ${commit} (tag-git run #${tagGitWorkflowRunID})`
+                const latest = runs
+                    .filter(run => run.output.summary === needle)
```

Comment:
This filtering feels fragile. Is this the best (only, I suspect 😢) way to filter runs?

Reply:
Yes, there is no other way, really. I use the `title`, `summary` and `output` to "store" information that links those workflow runs together.

Reply:
For the record: The `summary` is specified [in the `git-artifacts` workflow](https://github.com/git-for-windows/git-for-windows-automation/blob/9fe326b33ada8407e286daf131412986f7b878e9/.github/workflows/git-artifacts.yml#L128).